### ISSUE TITLE
Add System.Drawing.Design members

### ISSUE THREAD

**Issue by ericstj** [dotnet/corefx#29957]
The following were previously in System.Drawing and need to be brought back.

IPropertyValueUIService
PropertyValueUIHandler
PropertyValueUIItem
PropertyValueUIItemInvokeHandler
PaintValueEventArgs
UITypeEditor
UITypeEditorEditStyle

I've temporarily brought them back as a private copy, but they belong in CoreFx along-side other System.Drawing members, potentially in their own assembly.

**Comment by danmosemsft** [comment]
@ericstj any particular advantage of creating a new assembly over reusing System.Drawing.Common?

**Comment by ericstj** [comment]
It'll bring in a dependency on TypeConverter which I think @weshaggard would like to avoid.  That said Drawing.Common isn't part of shared framework so not sure how much we care.  I guess it just depends on if @weshaggard classifies Drawing.Common in the same group as TypeConverter.  My gut feel is that Drawing.Common was trying to peel away the app-model-isms from Drawing, and putting this in there adds some back.

This bit of registration is a layering concern:
https://github.com/dotnet/corefx/blob/master/src/System.Drawing.Common/src/System/Drawing/Design/UITypeEditor.cs#L34-L41
I think this still needs to live above those core types.  We may have a System.Design, or create some new assembly.  It seems at least somewhat likely that in either case we'll probably have some System.Design assembly (perhaps just forwards).  /cc @Tanya-Solyanik

**Comment by weshaggard** [comment]
I would like to keep Drawing decoupled from TypeConverter as much as possible as it includes a lot of dependencies. Depending on the exact dependency we can try to invert it but if we cannot I think we should consider a newer higher level library.

**Comment by danmosemsft** [comment]
@ericstj @merriemcgaw many of the concrete implementations of UITypeEditor are in System.ComponentModel.Design (this was in System.Design.dll). We don't have any of that namespace. I assume we consider that part of Winforms?

**Comment by merriemcgaw** [comment]
Yes, we should consider that part of WinForms. IIRC that is a WinForms designer type. I don't know if there are any non-VS scenarios for it. @Tanya-Solyanik should be able to give more details.

**Comment by ericstj** [comment]
Win forms runtime only needs to know about the base types I’ve listed in this issue.  The rest can remain in designer assemblies that ship in VS (or some TBD tooling component)

**Comment by safern** [comment]
Taking this one.
